Seeing this failure

```
tests/validator/test_v1_validator.py::test_validate_expectation_with_batch_asset_options[test_backends0]
```
```
>       assert result.success
E       assert False
E        +  where False = {\n  "success": false,\n  "expectation_config": {\n    "expectation_type": "expect_column_values_to_be_in_set",\n    "kwar..."exception_info": {\n    "raised_exception": false,\n    "exception_traceback": null,\n    "exception_message": null\n  }\n}.success
```

--- a/great_expectations/execution_engine/sqlalchemy_batch_data.py
+++ b/great_expectations/execution_engine/sqlalchemy_batch_data.py
@@ -96,7 +96,7 @@ class SqlAlchemyBatchData(BatchData):
         source_table_name: Optional[str] = None,
     ) -> None:
         """A Constructor used to initialize and SqlAlchemy Batch, create an id for it, and verify that all necessary
-        parameters have been provided. If a Query is given, also builds a temporary table for this query
+        parameters have been provided. Builds a temporary table for the `query` if `create_temp_table=True`.
 
             Args:
                 engine (SqlAlchemy Engine): \

--- a/great_expectations/execution_engine/sqlalchemy_execution_engine.py
+++ b/great_expectations/execution_engine/sqlalchemy_execution_engine.py
@@ -1360,22 +1360,21 @@ class SqlAlchemyExecutionEngine(ExecutionEngine):
         create_temp_table: bool = batch_spec.get(
             "create_temp_table", self._create_temp_table
         )
+        selectable: Union[
+            sqlalchemy.Selectable, str
+        ] = self._build_selectable_from_batch_spec(batch_spec=batch_spec)
         # NOTE: what's being checked here is the presence of a `query` attribute, we could check this directly
         # instead of doing an instance check
         if isinstance(batch_spec, RuntimeQueryBatchSpec):
             # query != None is already checked when RuntimeQueryBatchSpec is instantiated
-            query: str = batch_spec.query
 
             batch_data = SqlAlchemyBatchData(
                 execution_engine=self,
-                query=query,
+                selectable=selectable,
                 temp_table_schema_name=temp_table_schema_name,
                 create_temp_table=create_temp_table,
             )
         elif isinstance(batch_spec, SqlAlchemyDatasourceBatchSpec):
-            selectable: Union[
-                sqlalchemy.Selectable, str
-            ] = self._build_selectable_from_batch_spec(batch_spec=batch_spec)
             batch_data = SqlAlchemyBatchData(
                 execution_engine=self,
                 selectable=selectable,

--- a/tests/validator/test_v1_validator.py
+++ b/tests/validator/test_v1_validator.py
@@ -1,3 +1,7 @@
+from __future__ import annotations
+
+from pprint import pformat as pf
+
 import pytest
 
 from great_expectations.core.batch_config import BatchConfig
@@ -176,7 +180,7 @@ def test_validate_expectation_with_batch_asset_options(
             value_set=[desired_event_type],
         )
     )
-
+    print(f"Result dict ->\n{pf(result)}")
     assert result.success
 
 

